src/main/java/gateways/Tags/TagDatabaseGateway.java

```diff
@@ -0,0 +1,77 @@
+package gateways.Tags;
+
+import entities.Tag;
+import gateways.DatabaseGateway;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class TagDatabaseGateway extends DatabaseGateway implements TagDataAccessInterface {
+
+    private ArrayList<Tag> tagList;
+
+    /**
+     * Instantiating DatabaseGateway with a relative path will store the absolute path to the respective database.
+     * If the file at the path does not exist, one will be created
+     * This absolute path can be fetched using the getAbsoluteFilepath method
+     *
+     * @param relativePath The relative path to the database
+     * @throws IOException Throws exception if encountering failed or interrupted IO exceptions
+     */
+    public TagDatabaseGateway(String relativePath) throws IOException, ClassNotFoundException {
+        super(relativePath);
+        load();
+    }
+
+    @Override
+    public Tag get(int id) {
+        return null;
+    }
+
+    @Override
+    public ArrayList<Tag> getAll() {
+        return null;
+    }
+
+    @Override
+    public int insert(Tag tag) {
+        this.tagList.add(tag);
+        return 0;
+    }
+
+    @Override
+    public int update(Tag tag) {
```

Comment:
For update and delete, both threw errors because the DataAccessInterface required them to return booleans instead of integers. Why do they return integers here anyways? If the output must be integers, having them return false/true and then having a line of code in the method that handles them convert that to the desired integer (or maybe having a helper method) would probably take care of the error.